Task2/Kuba/NoughtsAndCrosses/GameBoard.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+
+namespace NoughtsAndCrosses
+{
+    public class GameBoard
+    {
+        public readonly int boardSize = 3; //static?
+        public GameBoardStatus[,] gameBoard;
+
+        public GameBoard()
+        {
+            gameBoard = new GameBoardStatus[boardSize, boardSize];
+        }
+
+        public void MarkPlayerTurn(Player player, Position coordinates)
+        {
+            if (CoordinatesAreValid(coordinates))
+            {
+                this.gameBoard[coordinates.Vertical, coordinates.Horizontal] = player.ChoosenSymbol;
+            }
+        }
+
+        public bool BoardIsFull()
+        {
+            for (int i = 0; i < boardSize; i++)
+            {
+                for (int j = 0; j < boardSize; j++)
+                {
+                    if (this.gameBoard[i, j] == GameBoardStatus.empty) return false;
+                }
+            }
+            return true;
+        }
+
+        public bool PlayerWon(Player player)
+        {
+            return (CheckVertically(player) || CheckHorizontally(player) || CheckDiagonally(player));
+        }
+
+        private bool CheckVertically(Player player)
+        {
+            for (int i = 0; i < boardSize; i++)
+            {
+                int playerSymbolCount = 0;
+                for (int j = 0; j < boardSize; j++)
+                {
+                    if (this.gameBoard[j, i] == player.ChoosenSymbol)
```

Comment:
We could also have one function instead of CheckVertically() and CheckHorizontally() by changing this line (the rest of code is identical for these two functions) or by having two 'playerSymbolCount'like variables.